implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/AwsAlbKeyConfigurationValidator.java

```diff
@@ -0,0 +1,72 @@
+package io.smallrye.jwt.auth.principal;
+
+import java.net.URI;
+
+import org.jose4j.lang.UnresolvableKeyException;
+
+interface AwsAlbKeyConfigurationValidator {
+
+    public static void validateKeyConfiguration(JWTAuthContextInfo authContextInfo) throws UnresolvableKeyException {
+        // public key location check
+        var publicKeyLocation = authContextInfo.getPublicKeyLocation();
+        if (publicKeyLocation == null) {
+            throw PrincipalMessages.msg.nullKeyLocation();
+        }
+        if (containsSubPath(publicKeyLocation)) {
+            throw AwsAlbKeyResolverMessages.msg.subPathNotAllowed();
+        }
+    }
+
+    public static void validatePublicKeyAlgorithmConfiguration(JWTAuthContextInfo authContextInfo) {
+        var publicKeyAlgorithm = authContextInfo.getSignatureAlgorithm();
+        if (publicKeyAlgorithm == null) {
+            AwsAlbKeyResolverLogging.log.publicKeyAlgorithmNotSet();
+        }
+        if (!publicKeyAlgorithm.getAlgorithm().equals("ES256")) {
```

Comment:
Please consider `SignatureAlgorithms.ES256.getAlgorithm()`

Reply:
@AdamBien I've prepared a tiny update to resolve this suggestion, are you OK if I push to your branch ?